pkg/resource/deploy/deployment.go

```diff
@@ -688,3 +693,30 @@ func (d *Deployment) Close() error {
 
 	return nil
 }
+
+func (d *Deployment) RunHooks(hookType resource.HookType, s *resource.State, canFail bool) error {
```

Comment:
`canFail` could do with some docs on what we expect here I think, it's not quite clear to me from just the name.  reading the code, I think what happens if that `canFail` means that it's a `before_*` hook, where we make the whole step fail if the hook fails?  Maybe `failStep` is a better name?